analyses/draw-sun-analysis/src/app.tsx

```diff
@@ -0,0 +1,57 @@
+import { Analysis } from "forma-embedded-view-sdk/analysis";
+import { Forma } from "forma-embedded-view-sdk/auto";
+import { useState } from "preact/hooks";
+import AnalysisPicker from "./components/AnalysisPicker";
+import TimePicker from "./components/TimePicker";
+import TriggerCalculation from "./components/TriggerCalcuation";
+
+function useSunAnalysis(projectId: string) {
+  const [analysis, setAnalysis] = useState<Analysis[] | undefined>();
+  const [proposalId, setProposalId] = useState<string>();
+  if (!proposalId) {
+    Forma.proposal.getId().then(setProposalId);
+  }
+  if (!analysis && proposalId) {
+    Forma.analysis
+      .list({
+        authcontext: projectId,
+      })
+      .then((analysis) => {
+        setAnalysis(
+          analysis.filter(
+            (analysis) =>
+              analysis.analysisType === "sun" && analysis.status === "SUCCEEDED" && analysis.proposalId === proposalId,
```

Comment:
Lets ask for an API to navigate to the correct proposal revision if we are not on the correct one.